lib/TraackrApi/Influencers.php

```diff
@@ -87,7 +87,12 @@ public static function addTwitter($p = array()) {
       $inf = new Influencers();
 
       $p = $inf->addCustomerKey($p);
-      $inf->checkRequiredParams($p, array('username', 'customer_key'));
+      $inf->checkRequiredParams($p, array('customer_key'));
+      
+      // Validate business requirements
+      if (!(empty($p['username']) xor empty($p['twitter_id']))) {
```

Comment:
I've never used the `xor` operator, but another way to state this is:

```php
// If both parameters are empty
// OR both parameters are not empty
if (empty($p['username']) == empty($p['twitter_id']))
```

Not sure if that's more understandable or not.

Or if you want to be more exact with error messages (this is the way I've done it in the API):

```php
if (empty($p['username']) && empty($p['twitter_id'])) {
  // Error: Either username or twitter_id must be present
} else if (!empty($p['username']) && !empty($p['twitter_id']) {
  // Error: Only one of username or twitter_id may be present
}
```